### ISSUE TITLE
Incorrect `maximumDate` cutoff on Android when `timeZoneOffsetInMinutes` is used

### ISSUE THREAD

**Issue by trooperandz** [henninghall/react-native-date-picker#613]
**Describe the bug**
Android issue only: When a `minimumDate` and a `maximumDate` range of 24 or 48 hours is passed in, and a `timeZoneOffsetInMinutes` is also passed in, the datetime picker does not show the correct max cutoff time available on Android. When `timeZoneOffsetInMinutes` is removed, the datetime picker correctly reflects the date range.

Note: The datetime picker worked correctly on an older Android device, with OS Android 9.

For instance:

- A date range of 24 hours (2/20 - 2/21) allows me to go from 2/20 4am up to 2/21 11:59pm (should be approximately 2/21 3:59am).
- A date range of 48 hours (2/20 - 2/22) allows me to go from 2/20 4am up to 2/21 11:59pm (should be approximately 2/22 3:59am).

**Expected behavior**
The datetime picker should correctly observe the provided `minimumDate` and `maximumDate` range on Android devices when `timeZoneOffsetInMinutes` is passed in.

**To Reproduce**

```javascript
const [date, setDate] = React.useState(new Date());

<DatePicker
  androidVariant="nativeAndroid"
  date={date}
  is24hourSource="locale"
  minimumDate={new Date("2023-02-20T11:00:00.000Z")}
  maximumDate={new Date("2023-02-22T10:00:00.000Z")}
  onDateChange={setDate}
  timeZoneOffsetInMinutes={-420}
/>;
```

**Smartphone (please complete the following information):**

- OS: Android Samsung s21 13.0, Android emulator 11.0 API R, Android emulator 12.0 API 31.
- React Native version 0.70.6
- react-native-date-picker version 4.2.5

#### Demo with incorrect 48 hour range, including timeZoneOffsetInMinutes

![android-datetimepicker](https://user-images.githubusercontent.com/14365371/220088761-f6e31fba-6ad2-480f-935a-14648e54a8ba.gif)

#### Demo with correct 48-hour range, excluding timeZoneOffsetInMinutes

![android-datetimepicker-no-offset](https://user-images.githubusercontent.com/14365371/220088814-edf09928-d4d4-405a-b91b-d5617e94bdde.gif)

**Comment by randhi-w** [comment]
Im having the same issue too but only when we pass today as the maximumDate with timeZoneOffsetInMinutes when the mode of the date picker is 'dateTime'. The date picker renders dates 6 months ahead

**Comment by henninghall** [comment]
Thanks for reporting this together with a great explanation of the problem @trooperandz. A fix for this is released in version 4.2.11.
